esp32/mpconfigport.h

```diff
@@ -176,7 +176,6 @@ extern const struct _mp_obj_module_t mp_module_network;
     { MP_OBJ_NEW_QSTR(MP_QSTR_random), (mp_obj_t)&mp_module_urandom }, \
     { MP_OBJ_NEW_QSTR(MP_QSTR_re), (mp_obj_t)&mp_module_ure }, \
     { MP_OBJ_NEW_QSTR(MP_QSTR_select), (mp_obj_t)&mp_module_uselect }, \
-    { MP_OBJ_NEW_QSTR(MP_QSTR_socket), (mp_obj_t)&mp_module_usocket }, \
```

Comment:
Ah, oops, I got confused with this one and thought I'd just made a mistake ... so I made it the same as network and things started working, then I forgot all about it.

Now, looking at it, I remember what's going on: with socket in WEAK_LINKS the __init__ function doesn't get called if it is imported as 'socket', only as 'usocket'.
```
>>> import socket
>>> import usocket
I (110348) modsocket: Initializing
```
This is caused by `py/objmodule.c:mp_module_get()` which only checks `mp_module_builtin_table` and not `mp_builtin_module_weak_links_table` when deciding if it should run the module's `__init__`.  Should I put in a PR for that against the main micropython project?